scripts/data/v3/infrastructure-damaged.ts

```diff
@@ -0,0 +1,147 @@
+import { ApiResource } from "../../../types/api.types";
+import { writeJson } from "../../utils/fs";
+import { SheetTab, fetchGoogleSheet } from "../../utils/gsheets";
+import {
+  formatDailiesJson,
+  validateDailiesJson,
+} from "../common/casualties-daily";
+
+const jsonFileName = "infrastructure-damaged.json";
+
+const columnFilter = new Set([
+  "report_date",
+  "civic_buildings.destroyed",
+  "ext_civic_buildings.destroyed",
+  "educational_buildings.destroyed",
+  "ext_educational_buildings.destroyed",
+  "educational_buildings.damaged",
+  "ext_educational_buildings.damaged",
+  "mosques.destroyed",
+  "ext_mosques.destroyed",
+  "mosques.damaged",
+  "ext_mosques.damaged",
+  "churches.destroyed",
+  "ext_churches.destroyed",
+  "units.destroyed",
+  "ext_units.destroyed"
+]);
+
+const generateJsonFromGSheet = async () => {
+  const sheetJson = await fetchGoogleSheet(SheetTab.InfrastructureDamaged);
+  const [, headerKeys, ...rows] = sheetJson.values;
+  
+  const jsonArray = formatInfrastructreJson(headerKeys, rows, columnFilter);
+  
+  validateJson(jsonArray);
+
+  writeJson(ApiResource.InfrastructureDamagedV3, jsonFileName, jsonArray);
+  console.log(`generated JSON file: ${jsonFileName}`);
+
+};
+
+const yyyymmddFormat = /^202[3-4]-[0-1][0-9]-[0-3][0-9]$/;
+
+/**
+ * our docs claim fields prefixed with ext_ are non-optional, so we should assert that
+ * we should also assert standard date format
+ */
+export const validateJson = (
+  json: Array<Record<string, any>>
+) => {
+  const uniqueFieldNames = new Set<string>();
+  json.forEach((record) => {
+    // validate date format is as expected for the base daily dataset format
+    const dateValid = yyyymmddFormat.test(record.report_date as string);
+    if (!dateValid) {
+      throw new Error(
+        `Report date '${record.report_date}' is invalid, expected YYYY-MM-DD`
+      );
+    }
+
+    // track all of the field names we use in the dataset
+    Object.keys(record).forEach((key) => uniqueFieldNames.add(key));
+  });
+  const extKeys = Array.from(uniqueFieldNames).filter((key) =>
+    key.startsWith("ext_")
```

Comment:
i recognize that the suggestion above to nest `ext_` requires updating the validation logic - i would suggest for now commenting the validation out until we're happy with the format and we can add it back in later (line 35)